File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/SmokeBombEffect.ts
import { SoundMaster } from "../../Audio/SoundMaster";
import DungeonManager from "../../DungeonManager";
import { AbstractEffect } from "../AbstractEffect";
import { SmokeBlurEffect } from "./SmokeBlurEffect";

export class SmokeBombEffect extends AbstractEffect{
    public constructor(private X,private Y) {
        super();
    }
    
    Update() {
        super.Update()
        if(this.Duration == 0){
            SoundMaster.PlayVoice("ATTACK_WHIFF_2");
            for(let i = 0;i<90;i++){
                DungeonManager.EffectManager.Play(new SmokeBlurEffect(this.X,this.Y));
            }
        }
        if(this.Duration >= 0.2)
        {
            SoundMaster.PlayVoice("ATTACK_WHIFF_2");
            this.IsDone = true;
        }
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/SmokeBombEffect.cs
using ModTheSpire.Dungeon;

namespace ModTheSpire.Effect.Combat
{
    public class SmokeBombEffect : AbstractEffect
    {
        float mX;
        float mY;

        public SmokeBombEffect(float x, float y)
        {
            mX = x;
            mY = y;
        }

        public override void Update()
        {
            base.Update();
                if(this.Duration == 0){
                    SoundMaster.PlayVoice(SoundMaster.ATTACK_WHIFF_2);
                    for(var i = 0;i<90;i++){
                        DungeonManager.EffectManager.Play(new SmokeBlurEffect(this.mX,this.mY));
                    }
                }
                if(this.Duration >= 0.2)
                {
                    SoundMaster.PlayVoice(SoundMaster.ATTACK_WHIFF_2);
                    Finish();
                }
        }
    }
}